### DISCUSSION TITLE
BB builder version update breaks App in older version

### DISCUSSION THREAD

**Discussion by micebrain** [Budibase/budibase#4138] [Help] [answered]
Hi,
The new December update looks great!
I updated the builder to 1.0.44 to checkout the new feature. However this breaks my app that is still in version 1.0.35
I can see that the app shows that new version is available, but I did not update the app.

<img width="336" alt="Screenshot 2022-01-21 at 07 23 23" src="https://user-images.githubusercontent.com/7136209/150477790-2f74db32-8b56-48f7-b5b4-1f5b3de61da6.png">

And I can see in the console that it is loading the client version 1.0.35

<img width="788" alt="Screenshot 2022-01-21 at 07 23 56" src="https://user-images.githubusercontent.com/7136209/150477842-eec51227-306c-47cf-895f-4e9693833a09.png">

**What breaks**

Bindings to variable disappeared in version 1.0.44
I have Rest API with bindings to other data provider which appears in the `Helpers` but now they disappeared

How does BB handles the different versions of app? Can I safely update the builder to check new features without updating the app?

**Comment by mjashanks** [comment, marked as answer]
Hey @micebrain 

If you want to try an upgrade, export your app, and re-import it (i.e. creating a new app).

I don't understand your issues, could you provide more detail or screenshots on the issues. I'm not sure what you mean by the following.
- Bindings to variable disappeared
- I have Rest API with bindings to other data provider

Happy to help if you give some more info :)

**Reply by micebrain** [reply]
Hello @mjashanks 
Thanks for the explanation on upgrading. Actually what happened was that I didn't upgrade the app yet, only the builder by clicking on the `Update` button. For some reasons, this affect the app without it being upgraded. 

<img width="556" alt="Screenshot 2022-01-21 at 19 42 36" src="https://user-images.githubusercontent.com/7136209/150582726-448eaf9d-53ba-4f67-90c8-76feff597551.png">
.
I started to get errors on some screens with Rest API and upon checking I discovered this scenario explained below.
In my app, I have a data provider with Rest API source and data displayed in table.
In the table, there is a link with Action - Execute Query with data source Rest. This API has a parameter defined.

<img width="793" alt="Screenshot 2022-01-21 at 19 50 30" src="https://user-images.githubusercontent.com/7136209/150583642-cdd11bdc-c0ba-461e-9f60-d9f8c62f6cc5.png">

When I click on the parameter binding button, I see that the available variables for bindings disappeared.  

<img width="623" alt="Screenshot 2022-01-21 at 19 36 57" src="https://user-images.githubusercontent.com/7136209/150583381-f357494d-9ecf-4043-8df3-cc0ed1e8f5f3.png">

**Reply by SantoshSrinivas79** [reply]
I seem to be having the same issue on `1.0.44` ... I am unable to find the form field bindings in the `Execute Query` button action.

It doesn't work on a completely new app either

**Reply by mjashanks** [reply]
Hi @SantoshSrinivas79 @micebrain  - I have tried this from a new app and an old app (pre v1). Both worked fine. My component setup is

```
> Data Provider : Internal table
   > Repeater
      > Button : Execute Query on a Rest API, passing parameters from above Data Provider
```

If either of you are able to export your app and send to support@budibase.com, i'd be happy to to have a look.

**Reply by mjashanks** [reply]
Just to let you know, I just encountered this issue also. Logged it here: https://github.com/Budibase/budibase/issues/4243 - we'll take a look.

**Reply by mjashanks** [reply]
@SantoshSrinivas79 it's actually already fixed on `1.0.46`

**Reply by micebrain** [reply]
Just picking up this thread again. Thanks @mjashanks!
